lib/predicate/registry.js

```diff
@@ -0,0 +1,367 @@
+import path from 'path';
+import domain from 'domain';
+import stream from 'stream';
+import { errors } from 'arsenal';
+import validate from './validate';
+
+const PredicateError = require('swerrs').extend({
```

Comment:
Fair point. The `swerrs` library just helps aggregate errors over the course of multiple operations, but it's optional; `validate` can be used with a regular array.